activities/ActivityTemplate/VueJS/js/components/SugarIcon.js

```diff
@@ -0,0 +1,47 @@
+Vue.component('sugar-icon', {
+	data: function () {
+		return {
+			icon: null
+		}
+	},
+	mounted() {
+		var vm = this;
+		requirejs(["sugar-web/graphics/icon"], function (icon) {
+			vm.icon = icon;
+		});
+	},
+	methods: {
+		colorize: function (element, colors, callback) {
+			this.icon.colorize(element, colors, callback);
+		},
+
+		generateIconWithColors: function (path, colors, callback) {
+			requirejs([`text!${path}`], function (icon) {
+				icon = icon.replace(/(stroke)_color\s\"#?\w*\"/, `stroke_color "${colors.stroke}"`);
+				icon = icon.replace(/(fill)_color\s\"#?\w*\"/, `fill_color "${colors.fill}"`);
+				callback("data:image/svg+xml;base64," + btoa(icon));
+			});
+		},
+
+		colorizeIcon: function (element, colors, callback) {
```

Comment:
What the difference between this method and the `colorize` method.
The name of methods are not obvious: `SugarIcon.colorize()` vs `SugarIcon.colorizeIcon()`...

Reply:
`colorize` is the wrapper for the legacy method `icon.colorize`, I thought about keeping it until the `colorizeIcon()` is perfect. Note that `colorizeIcon()` doesn't use GET request while `colorize()` does.